lib/ansible/plugins/action/__init__.py

```diff
@@ -146,7 +153,10 @@ def _configure_module(self, module_name, module_args, task_vars=None):
         # insert shared code and arguments into the module
         (module_data, module_style, module_shebang) = modify_module(module_name, module_path, module_args, task_vars=task_vars, module_compression=self._play_context.module_compression)
 
-        return (module_style, module_shebang, module_data)
+        if self._is_binary(module_path):
+            return ('non_native_want_json', None, None, module_path, True)
+
```

Comment:
This check (and _is_binary) seem like they belong in executor/module_common.py.  Make modify_module responsible for detecting that this is a binary module and returning it unchanged.  We might need to create a new module type so that the rest of the code in action/__init__.py knows it's dealing with a binary module but that shouldn't be a problem.  We already have "old" and "non_native_want_json" which do nearly the same things.